org.eclipse.lemminx/src/test/java/org/eclipse/lemminx/extensions/contentmodel/XMLSchemaDiagnosticsTest.java

```diff
@@ -643,6 +650,52 @@ public void testTargetNamespace_2SingleQuotes() throws Exception {
 		testCodeActionsFor(xml, targetNamespace, settings, ca(targetNamespace, te(2, 16, 2, 16, " xmlns='http://two-letter-name'")));
 	}
 
+	@Test
+	public void localSchemaFileMissingCodeAction() throws Exception {
+		String xml = "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n" + //
+				"<invoice xmlns:xsi=\"http://www.w3.org/2001/XMLSchema-instance\"\n" + //
+				"  xsi:noNamespaceSchemaLocation=\"/salad.xsd\">\n" + //
+				"</invoice>";
+		Diagnostic missingSchema = d(2, 32, 44, XMLSchemaErrorCode.schema_reference_4);
+		missingSchema.setMessage("schema_reference.4: Failed to read schema document "
+				+ "'file:///salad.xsd',"
+				+ " because 1) could not find the document; 2) the document could not be read;"
+				+ " 3) the root element of the document is not <xsd:schema>.");
+		Diagnostic eltDiagnostic = d(1, 1, 8, XMLSchemaErrorCode.cvc_elt_1_a);
+		eltDiagnostic.setMessage("cvc-elt.1.a: Cannot find the declaration of element 'invoice'.");
+		XMLAssert.testDiagnosticsFor(xml, missingSchema, eltDiagnostic);
+
+		SharedSettings settings = new SharedSettings();
+		WorkspaceClientCapabilities workspace = new WorkspaceClientCapabilities();
+		WorkspaceEditCapabilities workspaceEdit = new WorkspaceEditCapabilities();
+		workspaceEdit.setResourceOperations(Arrays.asList(ResourceOperationKind.Create));
+		workspace.setWorkspaceEdit(workspaceEdit);
+		settings.getWorkspaceSettings().setCapabilities(workspace);
+
+		XMLAssert.testCodeActionsFor(xml, missingSchema, settings,
+				ca(missingSchema,
+				createFile("/salad.xsd", false), 
+				teOp("/salad.xsd", 0, 0, 0, 0, "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n<xs:schema xmlns:xs=\"http://www.w3.org/2001/XMLSchema\">\n\n</xs:schema>")));

```

Comment:
I needed to change both occurrences of `"/salad.xsd"` to `"file:///salad.xsd"` in order to get this to pass under Linux. I think we need to adapt some of the test functions to work cross platform.

Reply:
indeed, in Windows OS too, fixed